### PR TITLE
Add missing error check after vm_op_get_value

### LINKED ISSUE #2654: Assertion 'ecma_is_value_undefined (value) || ecma_is_value_null (value) || ecma_is_value_boolean (value) || ecma_is_value_number (value) || ecma_is_value_string (value) || ecma_is_value_object (value)' failed in ecma_check_value_type_is_spec_defined
###### Jerry version:

```
Checked revision: ec142f5f
Build command: ./tools/build.py --clean --debug --profile=es2015-subset --logging=on
```

###### OS:

```
Ubuntu 18.04, x86_64
```

###### Test case:

```javascript
--null.static ? this : [ , ] instanceof true; 
```

###### Backtrace:

```
ICE: Assertion 'ecma_is_value_undefined (value) || ecma_is_value_null (value) || ecma_is_value_boolean (value) || ecma_is_value_number (value) || ecma_is_value_string (value) || ecma_is_value_object (value)' failed at jerryscript/jerry-core/ecma/base/ecma-helpers-value.c(ecma_check_value_type_is_spec_defined):386.
Error: ERR_FAILED_INTERNAL_ASSERTION

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
51	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff7a24801 in __GI_abort () at abort.c:79
#2  0x000055555555f2f6 in jerry_port_fatal (code=ERR_FAILED_INTERNAL_ASSERTION) at jerryscript/jerry-port/default/default-fatal.c:71
#3  0x000055555558b684 in jerry_fatal (code=ERR_FAILED_INTERNAL_ASSERTION) at jerryscript/jerry-core/jrt/jrt-fatals.c:63
#4  0x000055555558b6d6 in jerry_assert_fail (
    assertion=0x5555555e3c78 "ecma_is_value_undefined (value) || ecma_is_value_null (value) || ecma_is_value_boolean (value) || ecma_is_value_number (value) || ecma_is_value_string (value) || ecma_is_value_object (value)", file=0x5555555e3c28 "jerryscript/jerry-core/ecma/base/ecma-helpers-value.c", 
    function=0x5555555d0ac0 <__func__.3728.lto_priv.660> "ecma_check_value_type_is_spec_defined", line=386) at jerryscript/jerry-core/jrt/jrt-fatals.c:87
#5  0x00005555555b0570 in ecma_check_value_type_is_spec_defined (value=24) at jerryscript/jerry-core/ecma/base/ecma-helpers-value.c:381
#6  0x000055555559dd8c in ecma_op_to_number (value=24) at jerryscript/jerry-core/ecma/operations/ecma-conversion.c:244
#7  0x00005555555988a3 in vm_loop (frame_ctx_p=0x7fffffffdc00) at jerryscript/jerry-core/vm/vm.c:1819
#8  0x000055555559bcfe in vm_execute (frame_ctx_p=0x7fffffffdc00, arg_p=0x0, arg_list_len=0) at jerryscript/jerry-core/vm/vm.c:3428
#9  0x000055555559bfd7 in vm_run (bytecode_header_p=0x5555558f75c0 <jerry_global_heap+512>, this_binding_value=27, lex_env_p=0x5555558f73f0 <jerry_global_heap+48>, parse_opts=0, arg_list_p=0x0, 
    arg_list_len=0) at jerryscript/jerry-core/vm/vm.c:3531
#10 0x00005555555958f6 in vm_run_global (bytecode_p=0x5555558f75c0 <jerry_global_heap+512>) at jerryscript/jerry-core/vm/vm.c:227
#11 0x00005555555c2595 in jerry_run (func_val=203) at jerryscript/jerry-core/api/jerry.c:544
#12 0x00005555555c0ea0 in main (argc=3, argv=0x7fffffffe038) at jerryscript/jerry-main/main-unix.c:676
```

<sup>Found by [Fuzzinator](http://fuzzinator.readthedocs.io/) with [grammarinator](https://github.com/renatahodovan/grammarinator). </sup>

### PR DESCRIPTION
**Pull request by rerobika**
This patch fixes #2654.

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/vm/vm.c [hunk 1]
```diff
@@ -1733,6 +1733,10 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
 
             break;
           }
+          else if (ECMA_IS_VALUE_ERROR (result))
```
Comment (LaszloLango) [inline jerry-core/vm/vm.c]: The indentation is wrong (from line 1734 to 1742)

Comment (rerobika) [reply jerry-core/vm/vm.c]: I reordered the lines for better readability.

File: jerry-core/vm/vm.c [hunk 2]
```diff
@@ -1721,16 +1721,20 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
...
+              left_value = ECMA_VALUE_UNDEFINED;
+              right_value = ECMA_VALUE_UNDEFINED;
             }
+            goto error;
+          }
 
+          if (opcode < CBC_PRE_INCR)
+          {
+            left_value = ECMA_VALUE_UNDEFINED;
+            right_value = ECMA_VALUE_UNDEFINED;
```
Comment (akosthekiss) [inline jerry-core/vm/vm.c]: Is the proposed approach best way to write this?

```suggestion
if (opcode < CBC_PRE_INCR)
{
  left_value = ECMA_VALUE_UNDEFINED;
  right_value = ECMA_VALUE_UNDEFINED;
}

if (ECMA_IS_VALUE_ERROR (result))
{
  goto error;
}
	
if (opcode < CBC_PRE_INCR)
{
  break;
}
```
Wouldn't this snippet involve less copy-paste?

Comment (rerobika) [reply jerry-core/vm/vm.c]: Good idea! I've applied it.

File: jerry-core/vm/vm.c [hunk 3]
```diff
@@ -1725,12 +1725,15 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
           {
             left_value = ECMA_VALUE_UNDEFINED;
             right_value = ECMA_VALUE_UNDEFINED;
+          }
```
Comment (LaszloLango) [inline jerry-core/vm/vm.c]: The indentation is still off by two spaces from line 1725 to 1738.

Comment (rerobika) [reply jerry-core/vm/vm.c]: Am I missing something? I think the indentation is correct. All the 3 if statements must be at the same level with the

```c
  stack_top_p += 2;
  left_value = result;
  right_value = ECMA_VALUE_UNDEFINED;
  /* FALLTHRU */
```

part.

Comment (akosthekiss) [reply jerry-core/vm/vm.c]: The glitch is between lines 1724 and 1725.

Comment (rerobika) [reply jerry-core/vm/vm.c]: I see now! That's a github glitch, that I've already met several times before. See the related [part](https://github.com/rerobika/jerryscript/blob/fix_issue_2654/jerry-core/vm/vm.c#L1724) from my branch where the indentation is correct! :)

Comment (akosthekiss) [reply jerry-core/vm/vm.c]: Indeed. ("View file" also reveals that indentation is correct.)

**Pull request merged**